README.rst

```diff
@@ -150,33 +150,70 @@ To update page objects installed from external repos:
 How to run Whitelabel tests
 ----------------------------
 
-Before running whitelabel tests please ensure that these environment variables are set.
+
+1. Before running the commands change your working directory to `edx-e2e-tests`.
+
+.. code:: bash
+
+    cd edx-e2e-tests/
+
+2. Install the requirements
+
+.. code:: bash
+
+    pip install -r requirements/base.txt
+
+3. Install the page objects for the application from the edx platform repo. This will
+   clone the entire repo into lib/edx-platform so that it can use the page objects and
+   helper methods from common/test/acceptance. We also install capa and xmodule into the
+   virtual environment from common/lib.
+
+
+.. code:: bash
+
+    paver install_pages
+
+4. Set these Environment variables
 
 .. code:: bash
 
+    ==> TEST_ENV (Only required if tests are not running on stage, e.g. for sandbox just assign value `sandbox')
+    ==> TARGET_DNS (Only required if tests are not running on stage, just provide DNS e.g. just provide "helio" for "https://mitxpro-helio.sandbox.edx.org")
+    ==> ORG (Only valid values for now are "MITxPRO" and "HarvardMedGlobalAcademy", if not provided the tests wil run against MITxPRO by default)
     ==> BASIC_AUTH_USER
     ==> BASIC_AUTH_PASSWORD
-    ==> USER_LOGIN_EMAIL
-    ==> USER_LOGIN_PASSWORD
-    ==> COURSE_RUN
-    ==> COURSE_DISPLAY_NAME
-    ==> COURSE_NUMBER
-    ==> COURSE_ORG
     ==> GLOBAL_PASSWORD
-    ==> STAFF_USER_EMAIL
-    ==> TEST_EMAIL_SERVICE
-    ==> TEST_EMAIL_ACCOUNT
-    ==> TEST_EMAIL_PASSWORD
-    ==> ACCESS_TOKEN
+    ==> OAUTH_CLIENT_ID (Only required for stage or sandbox is not using the default id)
+    ==> OAUTH_CLIENT_SECRET (Only required for stage or sandbox is not using the default secret)
+    ==> COURSE_RUN (Optional, use only if you are using a value other than default)
+    ==> COURSE_DISPLAY_NAME (Optional, use only if you are using a value other than default)
+    ==> COURSE_NUMBER (Optional, use only if you are using a value other than default)
+    ==> COURSE_ORG (Optional, use only if you are using a value other than specified in ORG var)
+
```

Comment:
All env variables should have default values. I should only have to specify those that I want to override.
I'm getting errors about STAFF_USER_EMAIL, OAUTH_CLIENT_ID, OAUTH_CLIENT_SECRET not being set.
BASIC_AUTH_* and GLOBAL_PASSWORD should have defaults as well

Reply:
DO you have the latest code, I have already removed the check for STAFF_USER_EMAIL, OAUTH_CLIENT_ID, OAUTH_CLIENT_SECRET, BASIC_AUTH_* .

The only compulsory var right now is GLOBAL_PASSWORD

Reply:
but that was changed due to the change in logic, I would rather have any early warning about a required env rather than tests failing due to an incorrect value. Since the repo is public there is no way we could give a correct value for password in code